src/flag/model.rs

```diff
@@ -1,43 +1,58 @@
 pub(crate) use imap::types::Flag;
 use serde::ser::{Serialize, SerializeSeq, Serializer};
-use std::ops::Deref;
 
-// Serializable wrapper for `imap::types::Flag`
+use std::borrow::Cow;
+use std::ops::{Deref, DerefMut};
+use std::collections::HashSet;
 
-#[derive(Debug, PartialEq)]
-struct SerializableFlag<'f>(&'f imap::types::Flag<'f>);
+/// Serializable wrapper for `imap::types::Flag`
+#[derive(Debug, PartialEq, Eq, Clone)]
+struct SerializableFlag<'flag>(&'flag imap::types::Flag<'flag>);
 
-impl<'f> Serialize for SerializableFlag<'f> {
+impl<'flag> Serialize for SerializableFlag<'flag> {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        serializer.serialize_str(match self.0 {
-            Flag::Seen => "Seen",
-            Flag::Answered => "Answered",
-            Flag::Flagged => "Flagged",
-            Flag::Deleted => "Deleted",
-            Flag::Draft => "Draft",
-            Flag::Recent => "Recent",
-            Flag::MayCreate => "MayCreate",
-            Flag::Custom(cow) => cow,
-            _ => "Unknown",
-        })
-    }
+        where
+            S: Serializer,
+        {
+            serializer.serialize_str(match self.0 {
+                Flag::Seen => "Seen",
+                Flag::Answered => "Answered",
+                Flag::Flagged => "Flagged",
+                Flag::Deleted => "Deleted",
+                Flag::Draft => "Draft",
+                Flag::Recent => "Recent",
+                Flag::MayCreate => "MayCreate",
+                Flag::Custom(cow) => cow,
+                _ => "Unknown",
+            })
+        }
 }
 
-// Flags
-
-#[derive(Debug, PartialEq)]
-pub struct Flags<'f>(&'f [Flag<'f>]);
-
-impl<'f> Flags<'f> {
-    pub fn new(flags: &'f [imap::types::Flag<'f>]) -> Self {
-        Self(flags)
+/// This struct type includes all flags which belong to a given mail.
+/// It's used in the [`Msg.flags`] attribute field of the `Msg` struct.
+///
+/// [`Msg.flags`]: struct.Msg.html#structfield.flags
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Flags(HashSet<Flag<'static>>);
+
+impl Flags {
+    pub fn new<'new>(flags: &[imap::types::Flag<'new>]) -> Self {
+        Self(
+            flags
+            .iter()
+            .map(|flag| convert_to_static(flag).unwrap())
+            .collect::<HashSet<Flag<'static>>>(),
+            )
     }
 }
 
-impl<'f> ToString for Flags<'f> {
+// ===========
+// Traits
+// ===========
+// ------------------
+// Common traits
+// ------------------
```

Comment:
That's interesting! I'll take a look into this comment!

Reply:
Ok, so I finished it now. Hm... I like to write stuff in a row or group them. Like putting all traits in a row or all "helper functions". Is that considered as a bad practice as well? Because it's in my opinion nice to see it like that if you open the file:

![image](https://user-images.githubusercontent.com/50843046/125155162-4a05c680-e15e-11eb-9046-9d24facb8240.png)

Or is it not even allowed to use `// -- <section> --` ? :thinking: